tenets/codelingo/go/ticker-in-for-switch/codelingo.yaml

```diff
@@ -1,12 +1,39 @@
+funcs:
+  - name: extractRaw
+    type: resolver
+    body: |
+      function(rawCode) {
+        var splitCode = rawCode.split("\n")
+        var re = /case <-time\.After\(\w*/
+        var blank = splitCode[splitCode.length -1]
+        blank = blank.substring(0, blank.indexOf("}"))
+
+        for (var i = 0; i < splitCode.length; i++) {
+        	line = splitCode[i]
+        	if (line.match(re)) {
+            var start = line.indexOf("(")
+            var end = line.lastIndexOf(")")
+            var callArgs = line.substring(start+1, end)
+            var firstLine = "ticker := time.NewTicker(" + callArgs + ")\n"
+            splitCode[i] = line.replace(/case <-time\.After\(.*/, "case <-ticker.C:")
+            splitCode[0] = blank + splitCode[0]
+            var joinCode = splitCode.join("\n")
+            return firstLine + joinCode
+          }
+        }
+        return rawCode
+      }
```

Comment:
Can you please comment this function and justify why it is necessary - what bits cannot be done in CLQL?